File: components/retail-store-domain/src/main/java/com/vmware/data/demo/retail/store/domain/Beacon.java
package com.vmware.data.demo.retail.store.domain;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.beans.Transient;

@Data
@NoArgsConstructor
public class Beacon {

	private String uuid;
	private int major;
	private int minor;
	private String category;
	private boolean entrance;
	private boolean checkout;

    public String getKey()
    {
    	return new StringBuilder().append(uuid).append("|")
				.append(major).append("|")
				.append(minor).toString();
    }
}

File: components/retail-market-domain/src/main/kotlin/com/vmware/data/retail/store/domain/Beacon.kt
package com.vmware.data.retail.store.domain

data class Beacon(
    var uuid: String = "",
    var major: Int = 0,
    var minor: Int = 0,
    var category: String = "",
    var entrance: Boolean = false,
    var checkout: Boolean = false
)
{
    var key: String?
    get() = uuid;
    set(value)
    {
        if(value !=null)
            uuid = value
    };
}
